### ISSUE TITLE
Build with `py-evm` feature fails on Python3.11/MacOS

### ISSUE THREAD

**Issue by fjarri** [ethereum/eth-tester#248]
* Version: 0.7.0b1
* Python: 3.11.0
* OS: osx

### What was wrong?

`pip install eth-tester[py-evm]` fails when building `pysha3`:

```
Installing collected packages: pysha3
  DEPRECATION: pysha3 is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for pysha3 ... error
  error: subprocess-exited-with-error
  
  × Running setup.py install for pysha3 did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      running install
      ...
      In file included from Modules/_sha3/sha3module.c:20:
      Modules/_sha3/backport.inc:78:10: fatal error: 'pystrhex.h' file not found
      #include "pystrhex.h"
               ^~~~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
```

### How can it be fixed?

`py-evm` feature has the following dependencies:
```
    "py-evm": [
        # Pin py-evm to exact version, until it leaves alpha.
        # EVM is very high velocity and might change API at each alpha.
        "py-evm==0.6.0a1",
        "eth-hash[pysha3]>=0.1.4,<1.0.0;implementation_name=='cpython'",
        "eth-hash[pycryptodome]>=0.1.4,<1.0.0;implementation_name=='pypy'",
    ],
```
[`pysha3`](https://github.com/tiran/pysha3) seems to be abandoned. Is there some reason `pycryptodome` cannot be used universally?

**Comment by kclowes** [comment]
> [pysha3](https://github.com/tiran/pysha3) seems to be abandoned. Is there some reason pycryptodome cannot be used universally?

You're right, pysha3 is abandoned, but it is much faster than pycryptodome. eth-hash is now compatible with Python 3.11 as of v0.5.1, but looks like eth-tester still needs a little bit of work. It's on our list of priorities.

**Comment by fjarri** [comment]
Is this fixed? 0.8.0b3 seems to be working on Py3.11 for me.
